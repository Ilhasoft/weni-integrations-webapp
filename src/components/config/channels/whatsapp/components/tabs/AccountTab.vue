<template>
  <div class="account-tab">
    <div class="account-tab__content">
      <div class="account-tab__content__info">
        <div class="account-tab__content__info__account">
          <div class="account-tab__content__info__account__title">
            {{ $t('WhatsApp.config.account.title') }}
          </div>
          <div class="account-tab__content__info__account__name">
            <unnnic-icon-svg
              class="account-tab__content__info__account__name__icon"
              icon="single-neutral-2"
              scheme="neutral-cloudy"
              size="sm"
            />
            <div class="account-tab__content__info__account__name__text">
              {{ fieldHandler(appConfig.phone_number.display_name) }}
            </div>
          </div>
          <div class="account-tab__content__info__account__business">
            {{ wabaInfo.name }}
          </div>
        </div>
      </div>

      <div>
        <div
          v-for="(section, index) in accountSections"
          :key="index"
          class="account-tab__content__section"
        >
          <div class="account-tab__content__section__title">
            <span class="account-tab__content__section__title__name">
              {{ $t(`WhatsApp.config.${section.name}.title`) }}
            </span>
          </div>

          <div
            v-for="(field, i) in section.fields"
            :key="i"
            class="account-tab__content__section__field"
          >
            <div class="account-tab__content__section__field__key">
              {{ $t(field.label) }}
            </div>
            <div class="account-tab__content__section__field__value">
              {{ field.value }}
            </div>
          </div>
        </div>
      </div>
    </div>
    <unnnic-button
      class="account-tab__close-button"
      type="secondary"
      size="large"
      :text="$t('general.Close')"
      @click="emitClose"
    />
  </div>
</template>

<script>
  export default {
    name: 'AccountTab',
    props: {
      appInfo: {
        type: Object,
        default: /* istanbul ignore next */ () => {},
      },
    },
    methods: {
      emitClose() {
        this.$emit('close');
      },
      fieldHandler(field) {
        return field ?? `-`;
      },
    },
    computed: {
      wabaInfo() {
        return this.appInfo?.config?.waba ?? {};
      },
      appConfig() {
        return (
          this.appInfo?.config ?? {
            phone_number: {},
            certificate: null,
            default_template_language: null,
            consent_status: null,
          }
        );
      },
      accountSections() {
        return [
          {
            name: 'channel',
            status: 'green',
            fields: [
              {
                type: 'text',
                name: 'phone_number',
                label: 'WhatsApp.config.channel.fields.phone_number',
                value: this.fieldHandler(this.appConfig.phone_number.display_phone_number),
              },
              {
                type: 'text',
                name: 'whatsapp_display_name',
                label: 'WhatsApp.config.channel.fields.whatsapp_display_name',
                value: this.fieldHandler(this.appConfig.phone_number.display_name),
              },
              {
                type: 'text',
                name: 'default_language_for_templates',
                label: 'WhatsApp.config.channel.fields.default_language_for_templates',
                value: this.fieldHandler(this.appConfig.default_template_language),
              },
              {
                type: 'text',
                name: 'certificate',
                label: 'WhatsApp.config.channel.fields.certificate',
                value: this.appConfig.certificate ?? 'N/A',
              },
              {
                type: 'text',
                name: 'consent_status',
                label: 'WhatsApp.config.channel.fields.consent_status',
                value: this.fieldHandler(this.appConfig.consent_status),
              },
            ],
          },
          {
            name: 'business_account',
            status: 'yellow',
            fields: [
              {
                type: 'text',
                name: 'waba_name',
                label: 'WhatsApp.config.business_account.fields.waba_name',
                value: this.fieldHandler(this.wabaInfo.name),
              },
              {
                type: 'text',
                name: 'message_on_behalf_of',
                label: 'WhatsApp.config.business_account.fields.message_on_behalf_of',
                value: this.fieldHandler(this.wabaInfo.message_behalf_name),
              },
              {
                type: 'text',
                name: 'timezone_id',
                label: 'WhatsApp.config.business_account.fields.timezone_id',
                value: this.fieldHandler(this.wabaInfo.timezone),
              },
              {
                type: 'text',
                name: 'waba_id',
                label: 'WhatsApp.config.business_account.fields.waba_id',
                value: this.fieldHandler(this.wabaInfo.id),
              },
              {
                type: 'text',
                name: 'namespace',
                label: 'WhatsApp.config.business_account.fields.namespace',
                value: this.fieldHandler(this.wabaInfo.namespace),
              },
            ],
          },
        ];
      },
    },
  };
</script>

<style lang="scss" scoped>
  .account-tab {
    display: flex;
    flex-direction: column;
    height: 100%;

    &__content {
      display: flex;
      flex-direction: column;
      padding-right: $unnnic-spacing-inline-sm;
      overflow-x: hidden;
      flex: 1;

      &__info {
        display: flex;
        justify-content: space-between;
        flex-wrap: wrap;

        &__account,
        &__conversations {
          flex: 1;

          &__title {
            font-weight: $unnnic-font-weight-black;
            font-size: $unnnic-font-size-body-lg;
            line-height: $unnnic-font-size-body-lg + $unnnic-line-height-md;
            color: $unnnic-color-neutral-darkest;

            margin-bottom: $unnnic-spacing-stack-sm;
          }

          &__name {
            display: inline-flex;
            align-items: center;
            font-size: $unnnic-font-size-body-gt;
            line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
            color: $unnnic-color-neutral-cloudy;

            &__icon {
              margin-right: $unnnic-inline-nano;
            }
          }

          &__time {
            margin-bottom: $unnnic-spacing-stack-xs;
          }

          &__business,
          &__limit {
            font-size: $unnnic-font-size-body-gt;
            line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
            color: $unnnic-color-neutral-cloudy;
            margin-top: $unnnic-spacing-stack-xs;
          }
        }

        &__conversations {
          flex: 1;
        }
      }

      &__section {
        display: flex;
        flex-direction: column;
        flex-basis: 100%;
        gap: $unnnic-spacing-inline-sm;

        &__title {
          display: flex;
          margin-bottom: $unnnic-spacing-stack-nano;
          margin-top: $unnnic-spacing-stack-md;
          gap: $unnnic-inline-xs;

          &__name {
            font-weight: $unnnic-font-weight-black;
            font-size: $unnnic-font-size-body-lg;
            line-height: $unnnic-font-size-body-lg + $unnnic-line-height-md;
            color: $unnnic-color-neutral-darkest;
          }
        }

        &__field {
          display: inline-flex;

          &__key,
          &__value,
          &__edit {
            flex: 1;
            width: 50%;
            font-size: $unnnic-font-size-body-gt;
            line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
            color: $unnnic-color-neutral-cloudy;
            margin: auto 0;
          }

          &__edit {
            display: inline-flex;
            gap: $unnnic-spacing-inline-xs;
            align-items: center;

            &__input {
              height: 29px;
              max-width: 70%;

              ::v-deep .input {
                height: 29px;
              }
            }
          }
        }
      }
    }

    &__close-button {
      margin-top: $unnnic-spacing-inset-sm;
      align-self: flex-end;
      width: 50%;
      min-height: 3rem;
    }
  }
</style>
