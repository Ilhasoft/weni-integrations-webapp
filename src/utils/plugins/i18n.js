import Vue from 'vue';
import VueI18n from 'vue-i18n';

Vue.use(VueI18n);

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';
import es_es from '../../locales/es_es.json';

const languages = {
  'en-us': en,
  'pt-br': pt_br,
  es: es_es,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en-us',
  fallbackLocale: 'en-us',
  messages,
});

export default i18n;
