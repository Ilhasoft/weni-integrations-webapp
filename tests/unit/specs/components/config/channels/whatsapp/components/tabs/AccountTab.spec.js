import { shallowMount, createLocalVue } from '@vue/test-utils';
import AccountTab from '@/components/config/channels/whatsapp/components/tabs/AccountTab.vue';
import i18n from '@/utils/plugins/i18n';

const localVue = createLocalVue();
describe('whatsapp/components/tabs/AccountTab.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(AccountTab, {
      localVue,
      i18n,
      mocks: {
        $router: {
          push: jest.fn(),
        },
      },
      stubs: {
        UnnnicIconSvg: true,
        StatusIndicator: true,
        UnnnicButton: true,
        UnnnicInput: true,
      },
      propsData: {
        appInfo: {
          config: {
            phone_number: {
              display_phone_number: '+5511999999999',
              display_name: 'Number Name',
              default_template_language: 'pt-BR',
              certificate: 'certificateData',
              consent_status: 'status',
            },
            waba: {
              name: 'Waba name',
              message_behalf_name: 'on behalf name',
              timezone: 'America/Sao_Paulo',
              id: '123',
              namespace: '123123123123123',
            },
          },
        },
      },
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should be rendered properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('emitClose()', () => {
    it('should emit close event', async () => {
      expect(wrapper.emitted('close')).toBeFalsy();
      wrapper.vm.emitClose();
      expect(wrapper.emitted('close')).toBeTruthy();
    });
  });

  describe('appConfig', () => {
    it('should return app config attribute', () => {
      expect(wrapper.vm.appConfig).toBe(wrapper.vm.appInfo.config);
    });

    it('should return empty if appInfo is undefined', async () => {
      await wrapper.setProps({ appInfo: undefined });
      expect(wrapper.vm.appConfig).toEqual({
        phone_number: {},
        certificate: null,
        default_template_language: null,
        consent_status: null,
      });
    });
  });

  describe('wabaInfo', () => {
    it('should return waba config attribute', () => {
      expect(wrapper.vm.wabaInfo).toBe(wrapper.vm.appInfo.config.waba);
    });

    it('should return empty if appInfo is undefined', async () => {
      await wrapper.setProps({ appInfo: undefined });
      expect(wrapper.vm.wabaInfo).toEqual({});
    });
  });

  describe('navigateToTemplates()', () => {
    it('should change route to templates', () => {
      const spy = spyOn(wrapper.vm.$router, 'push');
      expect(spy).not.toHaveBeenCalled();
      wrapper.vm.navigateToTemplates();

      expect(spy).toBeCalledTimes(1);
      expect(spy).toHaveBeenCalledWith({
        path: `/apps/my/${wrapper.vm.appInfo.code}/${wrapper.vm.appInfo.uuid}/templates`,
      });
    });
  });

  describe('navigateToCatalogs()', () => {
    it('should change route to catalogs', async () => {
      await wrapper.setProps({ hasCatalog: true });
      const spy = spyOn(wrapper.vm.$router, 'push');
      expect(spy).not.toHaveBeenCalled();
      wrapper.vm.navigateToCatalogs();

      expect(spy).toBeCalledTimes(1);
      expect(spy).toHaveBeenCalledWith({
        path: `/apps/my/${wrapper.vm.appInfo.code}/${wrapper.vm.appInfo.uuid}/catalogs`,
      });
    });

    it('should open facebook in a new tab', async () => {
      window.open = jest.fn();
      const spy = spyOn(wrapper.vm.$router, 'push');
      expect(spy).not.toHaveBeenCalled();
      expect(window.open).not.toHaveBeenCalled();
      wrapper.vm.navigateToCatalogs();

      expect(spy).not.toHaveBeenCalled();
      expect(window.open).toHaveBeenCalledTimes(1);
    });
  });
});
