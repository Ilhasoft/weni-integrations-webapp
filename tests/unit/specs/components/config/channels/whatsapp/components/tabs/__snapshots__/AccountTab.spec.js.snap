// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`whatsapp/components/tabs/AccountTab.vue should be rendered properly 1`] = `
<div class="account-tab">
  <div class="account-tab__content">
    <div class="account-tab__content__info">
      <div class="account-tab__content__info__templates">
        <div class="account-tab__content__info__templates__buttons">
          <div class="account-tab__content__info__templates__buttons__title">
            Templates
          </div>
          <unnnicbutton-stub type="alternative" size="small" scheme="feedback-green" class="account-tab__content__info__templates__buttons__button">
            See my templates
          </unnnicbutton-stub>
        </div>
        <div class="account-tab__content__info__templates__buttons">
          <div class="account-tab__content__info__templates__buttons__title">
            Catalog
          </div>
          <unnnicbutton-stub type="primary" size="small" class="account-tab__content__info__templates__buttons__button">
            View list of catalogs
          </unnnicbutton-stub>
        </div>
      </div>
      <div class="account-tab__content__info__qr">
        <div class="account-tab__content__info__qr__title">
          WhatsApp Link
        </div>
        <div class="account-tab__content__info__qr__wrapper"><img src="https://api.qrserver.com/v1/create-qr-code/?size=74x74&amp;data=https://wa.me/5511999999999" class="account-tab__content__info__qr__img">
          <div class="account-tab__content__info__qr__content"><span class="account-tab__content__info__qr__content__info">To access the link, scan the QR Code or click the button below</span>
            <unnnicbutton-stub type="secondary" text="Access link" iconleft="export-1" size="small" class="account-tab__content__info__qr__content__button"></unnnicbutton-stub>
          </div>
        </div>
      </div>
    </div>
    <div class="account-tab__content__section">
      <div class="account-tab__content__section__title"><span class="account-tab__content__section__title__name">
          WhatsApp Channel
        </span></div>
      <div class="account-tab__content__section__field">
        <div class="account-tab__content__section__field__key">
          Phone number
        </div>
        <div class="account-tab__content__section__field__value">
          +5511999999999
        </div>
      </div>
      <div class="account-tab__content__section__field">
        <div class="account-tab__content__section__field__key">
          WhatsApp Display Name
        </div>
        <div class="account-tab__content__section__field__value">
          Number Name
        </div>
      </div>
      <div class="account-tab__content__section__field">
        <div class="account-tab__content__section__field__key">
          Default Language for templates
        </div>
        <div class="account-tab__content__section__field__value">
          -
        </div>
      </div>
      <div class="account-tab__content__section__field">
        <div class="account-tab__content__section__field__key">
          Certificate
        </div>
        <div class="account-tab__content__section__field__value">
          N/A
        </div>
      </div>
      <div class="account-tab__content__section__field">
        <div class="account-tab__content__section__field__key">
          Consent Status
        </div>
        <div class="account-tab__content__section__field__value">
          -
        </div>
      </div>
    </div>
    <div class="account-tab__content__section">
      <div class="account-tab__content__section__title"><span class="account-tab__content__section__title__name">
          WhatsApp Business Account (WABA)
        </span></div>
      <div class="account-tab__content__section__field">
        <div class="account-tab__content__section__field__key">
          WABA Name
        </div>
        <div class="account-tab__content__section__field__value">
          Waba name
        </div>
      </div>
      <div class="account-tab__content__section__field">
        <div class="account-tab__content__section__field__key">
          Message on Behalf Name
        </div>
        <div class="account-tab__content__section__field__value">
          on behalf name
        </div>
      </div>
      <div class="account-tab__content__section__field">
        <div class="account-tab__content__section__field__key">
          Timezone ID
        </div>
        <div class="account-tab__content__section__field__value">
          America/Sao_Paulo
        </div>
      </div>
      <div class="account-tab__content__section__field">
        <div class="account-tab__content__section__field__key">
          WABA ID
        </div>
        <div class="account-tab__content__section__field__value">
          123
        </div>
      </div>
      <div class="account-tab__content__section__field">
        <div class="account-tab__content__section__field__key">
          Namespace
        </div>
        <div class="account-tab__content__section__field__value">
          123123123123123
        </div>
      </div>
    </div>
  </div>
</div>
`;
